File: basic-regex-paser/BasicRegexPaser.js
function isMatch(text, pattern) {
  // your code goes here

  if (pattern.length === 0) return text.length === 0;

  let match = false;
  if(text.length > 0){
    if (text[0] === pattern[0] || pattern[0] === '.')
      match = true;
  }

  if (pattern.length >= 2 && pattern[1] === '*') {
    return isMatch(text, pattern.substring(2)) || (match && isMatch(text.substring(1), pattern));
  }

  else return match && isMatch(text.substring(1), pattern.substring(1));
};

console.log(isMatch('abbb', 'ab*'));
// true
console.log(isMatch('mississippi', 'mis*is*p*.'));
// false
console.log(isMatch('acd', 'ab*c.'));
// true

File: basic-regex-paser/BasicRegexPaser.rb
def is_match(s, p)
  return true if /^#{p}$/.match(s)
  return false
end

puts is_match('abbb', 'ab*');
#  true
puts is_match('mississippi', 'mis*is*p*.');
#  false
puts is_match('acd', 'ab*c.');
#  true